libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -182,6 +186,8 @@ proc connectOnce(p: PubSubPeer): Future[void] {.async.} =
 
     await handle(p, newConn)
   finally:
+    if not p.connectedFut.finished():
+      p.connectedFut.fail(newException(LPError, "can't establish conn"))
```

Comment:
Is it possible to create a test?

Reply:
Testing the race condition is a bit tricky as it's quite random
Though when we'll merge 807, it will be a lot more frequent (but still random)

Reply:
Should it happen if we repeat it a large enough number of times?

Reply:
I'll do a better fix & try to create a test by manually slowing down the connection